ampd/src/broadcaster_v2/msg_queue.rs

```diff
@@ -100,13 +100,19 @@ where
         &mut self,
         msg: Any,
     ) -> Result<impl Future<Output = Result<(String, u64)>> + Send> {
-        let rx = self.enqueue_with_channel(msg).await?;
+        let attachment = json!({ "msg": msg });
+        let rx = self
+            .enqueue_with_channel(msg.clone())
+            .await
+            .map_err(|err| err.attach_printable(attachment.clone()))?;
 
-        Ok(rx.map(|result| match result {
-            Ok(Ok(result)) => Ok(result),
-            Ok(Err(err)) => Err(err),
-            Err(err) => Err(err.into_report()),
-        }))
+        Ok(rx
+            .map(|result| match result {
+                Ok(Ok(result)) => Ok(result),
+                Ok(Err(err)) => Err(err),
+                Err(err) => Err(err.into_report()),
+            })
+            .map_err(move |err| err.attach_printable(attachment)))
     }
```

Comment:
is this possible so we don't need to clone all over the place?
```
        let with_attachment = |err| err.attach_printable(json!({ "msg": msg }));
        let rx = self
            .enqueue_with_channel(msg.clone())
            .await
            .map_err(with_attachment)?;

        Ok(rx
            .map(|result| match result {
                Ok(Ok(result)) => Ok(result),
                Ok(Err(err)) => Err(err),
                Err(err) => Err(err.into_report()),
            })
            .map_err(with_attachment)
```

Reply:
That doesn't work, but I figured something out.